### ISSUE TITLE
Welcome the new highfive?

### ISSUE THREAD

**Issue by wafflespeanut** [servo/servo#13500]
About 2 weeks back, I [rewrote](http://github.com/servo-automation/highfive) the handlers of highfive for more control on how we handle Github webhook payloads. I have some ideas about how we should start moving stuff to the new highfive.
- It supports per-repo, per-handler and per-event configuration, and it could be easily setup on heroku (even the free dyno hours will serve us fine). We should set it up, create a new webhook for it, and begin _moving_ the events one by one from the old webhook to the new one.
- It supports multiple bots to share the API requests. So, whenever we hit the API limit, we could add a bot (like, one for each member of the organization) and handover the auth tokens to someone who'll be maintaining the heroku app (or wherever we're planning to host the server).

Once we begin, we'll setup a meta bug and start porting?

**Comment by aneeshusa** [comment]
I'm not sure adding yet-another-infra-provider (Heroku) is necessarily a good idea; we may want to run this with our other infra. Especially with multiple bots, we'll want configuration management to keep them in sync, so we may as well add it to saltfs.

**Comment by aneeshusa** [comment]
@Wafflespeanut, I appreciate the work you've done on this new highfive and there are some cool new features, but I personally have a few concerns about putting this into production at the moment:
- I'd rather not write/deploy new Python code that is Python2 only; IMO we should use Python3 for new code.
- The usage of git submodules for referencing handlers is worrying; in my experience git submodules always end up causing more pain than benefit. I also don't understand the benefits of using submodules in this case.
- Most importantly, I'm not aware of any major problems (aside from hitting rate limits) with highfive right now, and architecturally I think it would be better to have a highfive rewrite be a [GitHub integration](https://developer.github.com/early-access/integrations/) instead of a webhook handler (cc https://github.com/servo-highfive/highfive/issues/1). I think this is something worth waiting for, especially since it promises to have a cleaner way to handle hitting the rate limit than having multiple accounts/tokens (random usage can also easily lead to non-uniform/bad usage patterns, e.g. the Heroku/RapGenius random vs. intelligent routing fiasco). This may also obviate the need to run multiple instances.

A feature that would definitely help with rate limits would be GitHub API response caching; it's unclear to me whether the new highfive has this or not. I definitely like the better granularity of the configuration and ability to specify dependencies (like flake8), and I'm excited to be able to use a revamped highfive after some more work :)

**Comment by Wafflespeanut** [comment]
Thanks for the comments.

 - Of course! We can easily port it to python3
-  I've no idea why I'd decided to go for submodules in the first place. I remember talking to @jdm about it when I began this. I got carried away with it, and now, I'm not sure. I think we should just maintain a repo for this, instead of an organization (I admit, the workflow was relatively painful).
- Yeah, integration is definitely what we want. I'll look into setting it up for highfive.

Regarding caching, I *did* have an idea to maintain a simple database for storing the payloads (at least until the corresponding issues/PRs get closed), but I procrastinated it :smile:
